lambdas/handlers/delete_document_reference_handler.py

```diff
@@ -25,38 +25,70 @@
 )
 def lambda_handler(event, context):
     nhs_number = event["queryStringParameters"]["patientId"]
-    doc_type = event["queryStringParameters"]["docType"]
+    doc_type = extract_document_type(event["queryStringParameters"]["docType"])
 
     document_service = DocumentService()
 
     try:
-        table = ""
-        if SupportedDocumentTypes.ARF.name == doc_type:
-            logger.info("Retrieving ARF documents for deletion")
-            table = os.environ["DOCUMENT_STORE_DYNAMODB_NAME"]
+        if doc_type == SupportedDocumentTypes.ALL.value:
+            arf_delete_response = handle_delete(
+                document_service, nhs_number, str(SupportedDocumentTypes.ARF.value)
+            )
+            lg_delete_response = handle_delete(
+                document_service, nhs_number, str(SupportedDocumentTypes.LG.value)
+            )
 
-        if SupportedDocumentTypes.LG.name == doc_type:
-            logger.info("Retrieving Lloyd George documents for deletion")
-            table = os.environ["LLOYD_GEORGE_DYNAMODB_NAME"]
+            if (
+                arf_delete_response["statusCode"] == 404
+                and lg_delete_response["statusCode"] == 404
+            ):
+                return ApiGatewayResponse(
+                    404, "No documents available", "DELETE"
+                ).create_api_gateway_response()
```

Comment:
just a little suggestion, would it be good to change `extract_document_type` to return the enum type instead of enum.value string ? 
so that we could use directly use doc_type as our custom enum type within our boundary, and don't need to do a .value every time when we compare
Although I've read your tests cases and I think that current implementation should be working fine :) 

Or, maybe we can put this as a little side note for future refactoring 
(possibly after we bump up github action to python 3.11, when we can use StrEnum)

Reply:
Yeah I agree think this may need revisiting later on down the line as you've mentioned StrEnums would be great to implement across the whole app not just for SupportedDocTypes. I've added this to my refactoring notes